ipykernel/kernelbase.py

```diff
@@ -308,7 +312,14 @@ def dispatch_shell(self, msg):
             except Exception:
                 self.log.debug("Unable to signal in pre_handler_hook:", exc_info=True)
             try:
-                yield gen.maybe_future(handler(self.shell_stream, idents, msg))
+                result = handler(self.shell_stream, idents, msg)
+                if inspect.isawaitable(result):
+                    await result
+                else:
+                    warnings.warn(
+                    "Message handlers should be awaitable",
```

Comment:
I don't think there's a reason to deprecate this. It's fine and in fact preferable a lot of the time for these to not be async when they don't need to be.